mlir/test/Dialect/Affine/parallelize.mlir

```diff
@@ -0,0 +1,146 @@
+// RUN: mlir-opt %s -affine-parallelize | FileCheck %s
+
+// for multiple nested for-loops 
+#map0 = affine_map<(d0, d1, d2, d3) -> (d0, d1, d2, d3)>
+#map1 = affine_map<() -> (0)>
+#map2 = affine_map<() -> (64)>
+#map3 = affine_map<() -> (8)>
+#map4 = affine_map<() -> (1)>
+#map5 = affine_map<(d0, d1) -> (d0 + d1)>
+#map6 = affine_map<(d0, d1) -> (d0 * 2 + d1)>
+#map7 = affine_map<() -> (3)>
+// CHECK-DAG: [[MAP0:#map[0-9]+]] = affine_map<(d0, d1, d2, d3) -> (d0, d1, d2, d3)>
+// CHECK-DAG: [[MAP1:#map[0-9]+]] = affine_map<() -> (0)>
+// CHECK-DAG: [[MAP2:#map[0-9]+]] = affine_map<() -> (64)>
+// CHECK-DAG: [[MAP0:#map[0-9]+]] = affine_map<() -> (8)>
+// CHECK-DAG: [[MAP1:#map[0-9]+]] = affine_map<() -> (1)>
+// CHECK-DAG: [[MAP2:#map[0-9]+]] = affine_map<(d0, d1) -> (d0 + d1)>
+// CHECK-DAG: [[MAP0:#map[0-9]+]] = affine_map<(d0, d1) -> (d0 * 2 + d1)>
+// CHECK-DAG: [[MAP1:#map[0-9]+]] = affine_map<() -> (3)>
+// CHECK-LABEL:    func @reduce_window_max(
+func @reduce_window_max() {
+// CHECK-SAME:      ) {
+  %cst = constant 0.000000e+00 : f32
+  %0 = alloc() : memref<1x8x8x64xf32>
+  %1 = alloc() : memref<1x18x18x64xf32>
+  %2 = alloc() : memref<f32>
+  %3 = alloc() : memref<1x3x3x1xi32>
+  affine.for %arg0 = 0 to 1 {
+    affine.for %arg1 = 0 to 8 {
+      affine.for %arg2 = 0 to 8 {
+        affine.for %arg3 = 0 to 64 {
+          affine.store %cst, %0[%arg0, %arg1, %arg2, %arg3] : memref<1x8x8x64xf32>
+        }
+      }
+    }
+  }
+  affine.for %arg0 = 0 to 1 {
+    affine.for %arg1 = 0 to 8 {
+      affine.for %arg2 = 0 to 8 {
+        affine.for %arg3 = 0 to 64 {
+          affine.for %arg4 = 0 to 1 {
+            affine.for %arg5 = 0 to 3 {
+              affine.for %arg6 = 0 to 3 {
+                affine.for %arg7 = 0 to 1 {
+                  %4 = affine.apply #map5(%arg0, %arg4)
```

Comment:
Avoid using affine.apply's. Just run -canonicalize on this to get rid of affine.apply's.